lib/io_adapters.py

```diff
@@ -0,0 +1,100 @@
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+import numpy as np
+
+
+
+def create_input_adapter(input_shape, size=16, depth=40, activation=None):
+  """Creates an input adapter module for the input image.
+  The input adapter transforms input image of given shape 
+  into a tensor of target shape.
+
+  Arguments: 
+    input_shape: shape of input image (HxWxC). Image width and height 
+      must be devidible by size. H*W*C must be less than or equal
+      to size*size*depth.
+
+    size: height and width of the output tensor after space2depth operation. 
+
+    depth: number of channels in the output tensor.
+
+    activation: conv layer activation function"""
+  def _get_parameters():
+    h, w, c = input_shape
+    if h % size != 0 or  w % size != 0:
+      raise ValueError(
+          'input height and width should be devisible by output size.')
+    num_out_pixels = size * size * depth
+    num_in_pixels = h * w * c
+    if num_in_pixels >= num_out_pixels:
+      raise ValueError(
+          'input H*W*C should not be smaller than output H*W*C.')
+
+    # `block_size` of space2depth
+    block_size = min(h / size, w / size)
+    if depth % (block_size * block_size) != 0:
+      raise ValueError('depth value is not devivible by '
+      'the computed block size')   
```

Comment:
Style: Here and later make sure to make function arguments follow the open bracket.
```suggestion
                       'the computed block size')   
```